spinnaker/spinnaker_system/kube_find_image_test.py

```diff
@@ -0,0 +1,519 @@
+# limitations under the License.
+
+"""
+Integration test to see if the image promotion process is working for the
+Spinnaker Kubernetes integration.
+
+See testable_service/integration_test.py and spinnaker_testing/spinnaker.py
+for more details.
+
+The test will use ssh to peek at the spinnaker configuration
+to determine the managed project it should verify, and to determine
+the spinnaker account name to use when sending it commands.
+
+Sample Usage:
+    Assuming you have created $PASSPHRASE_FILE (which you should chmod 400)
+    and $CITEST_ROOT points to the root directory of this repository
+    (which is . if you execute this from the root)
+
+  PYTHONPATH=$CITEST_ROOT:$CITEST_ROOT/spinnaker \
+    python $CITEST_ROOT/spinnaker/spinnaker_system/kube_find_image_test.py \
+    --gce_ssh_passphrase_file=$PASSPHRASE_FILE \
+    --gce_project=$PROJECT \
+    --gce_zone=$ZONE \
+    --gce_instance=$INSTANCE
+or
+  PYTHONPATH=$CITEST_ROOT:$CITEST_ROOT/spinnaker \
+    python $CITEST_ROOT/spinnaker/spinnaker_system/kube_find_image_test.py \
+    --native_hostname=host-running-find-image-test
+"""
+
+# Standard python modules.
+import sys
+
+# citest modules.
+import citest.kube_testing as kube
+import citest.json_contract as jc
+import citest.json_predicate as jp
+import citest.service_testing as st
+
+# Spinnaker modules.
+import spinnaker_testing as sk
+import spinnaker_testing.gate as gate
+
+
+class KubeFindImageTestScenario(sk.SpinnakerTestScenario):
+  """Defines the scenario for the find image test.
+
+  This scenario defines the different test operations.
+  We're going to:
+    Create a Spinnaker Application
+    Create a Spinnaker Load Balancer
+    Create a Spinnaker Server Group
+    Create a Pipeline with the following stages
+      - Find Image
+      - Deploy
+      - Disable Server Group
+    Delete each of the above (in reverse order)
+  """
+
+  @classmethod
+  def new_agent(cls, bindings):
+    """Implements citest.service_testing.AgentTestScenario.new_agent."""
+    agent = gate.new_agent(bindings)
+    agent.default_max_wait_secs = 70
+    return agent
+
+  @classmethod
+  def initArgumentParser(cls, parser, defaults=None):
+    """Initialize command line argument parser.
+
+    Args:
+      parser: argparse.ArgumentParser
+    """
+    super(KubeFindImageTestScenario, cls).initArgumentParser(
+        parser, defaults=defaults)
+
+  def __init__(self, bindings, agent=None):
+    """Constructor.
+
+    Args:
+      bindings: [dict] The data bindings to use to configure the scenario.
+      agent: [GateAgent] The agent for invoking the test operations on Gate.
+    """
+    super(KubeFindImageTestScenario, self).__init__(bindings, agent)
+    bindings = self.bindings
+
+    optional_stack = ('-{0}'.format(bindings['TEST_STACK'])
+                      if bindings['TEST_STACK']
+                      else '')
+
+    # No detail because name length is restricted too much to afford one!
```

Comment:
I think it should even live in it's own module "frigga" which we can add under the spinnaker specific code, since Gate isn't really in charge of generating those names in the first place.